Tests/FluentAssertions.Specs/Primitives/ReferenceTypeAssertionsSpecs.cs

```diff
@@ -42,8 +42,8 @@ public void When_two_different_objects_are_expected_to_be_the_same_it_should_fai
         act
             .Should().Throw<XunitException>()
             .WithMessage(
-                "Expected subject to refer to*{ UserName = JohnDoe } because " +
-                "they are the same, but found*{ Name = John Doe }.");
+                "Expected subject to refer to*{*UserName =*JohnDoe*} because " +
```

Comment:
Are the spaces removed or additional characters added?

Reply:
The anonymous class that this test happens to use is now formatted over multiple lines, because it no longer uses the default `ToString` of generated types (anonymous classes, records and tuples) but instead writes out the member names and values.  This is now the same as normal classes that do not have a custom `ToString` implemented.  A side-effect of this is that there are newline characters between the braces and the properties.

Previously, the anonymous class `ToString` output 
```
Expected subject to refer to { UserName = JohnDoe } because they are the same, but found { Name = John Doe }.
```

Now that anonymous classes are formatted by FluentAssertions instead of the compiler generated `ToString`, the desired output is:
```
Expected subject to refer to
{
    UserName = "JohnDoe"
} because they are the same, but found
{
    Name = "John Doe"
}.
```

That asterisk before `JohnDoe` was a mistake and I'm glad you questioned this as it highlighted two problems - it was putting a newline between the property name and value; and it was *not* putting a newline before the opening brace:
```
Expected subject to refer to {
    UserName = 
    "JohnDoe"
} because they are the same, but found {
    Name = 
    "John Doe"
}." does not.
```

The brace was easy, but to solve the unwanted newline after the property name I've had to remove the `UseLineBreaks` support from `StringValueFormatter`.  All the tests still pass, and it does seem to follow how all the primitive formatters output their values.  I'm a little cautious that there are knock-on effects I cannot see.  Feedback on whether that's a good idea or not please.

Reply:
I myself do not like these line breaks. They interrupt the sentence structure.
At least "small" types should not get line breaks.
But this is just my personal opinion.

Reply:
That seems to be how FluentAssertions deals with writing out the properties of classes that don't explicitly have a `ToString`, unless I'm missing something?  

The compiler's implementation of anonymous classes' `ToString` happened to write all the properties on one line, FluentAssertions writes them out one per line.

All I've really done is treat anonymous types and records the same way as user-defined classes without `ToString`.  The message will be formatted the same now for user-defined types and compiler generated type.

Reply:
Is that something you'd like to be done as part of this change?  As it stands, this change simply formats compiler-generated types the same way that user-defined types have been generated for ages (all along?)

Reply:
Nah, it's a separate improvement, so let's keep the PR focused.